hooks/mutations/b2b/quotes/useUpdateQuoteCoupon/useUpdateQuoteCoupon.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * @module useUpdateQuoteCoupon
+ */
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+
+import { makeGraphQLClient } from '@/lib/gql/client'
+import { updateQuoteCouponMutation } from '@/lib/gql/mutations'
+import { quoteKeys } from '@/lib/react-query/queryKeys'
+
+interface UpdateQuoteCouponParams {
+  quoteId: string
+  couponCode: string
+}
+
+const updateQuoteCoupon = async (params: UpdateQuoteCouponParams) => {
+  const client = makeGraphQLClient()
+  const { quoteId, couponCode } = params
+
+  const variables = {
+    quoteId,
+    couponCode,
+    updateMode: 'ApplyToDraft',
+  }
+
+  const response = await client.request({
+    document: updateQuoteCouponMutation,
+    variables,
+  })
+
+  return response?.updateQuoteCoupon
+}
+
+/**
+ * [Mutation hook] useUpdateQuoteCoupon uses the graphQL mutation
+ *
+ * <b>updateQuoteCoupon(quoteId: String!, couponCode: String!): Quote</b>
+ *
+ * Description : Applies promo code in Quote Summary of quote details pages
+ *
+ * Parameters passed to function updateQuoteCoupon(params: UpdateQuoteCouponParams) => expects object of type 'UpdateQuoteCouponParams' containing quoteId, updateMode and couponCode
```

Comment:
parameters are miss matched with  <b>updateQuoteCoupon(quoteId: String!, couponCode: String!): Quote</b>, please verify and also check for deleteQuoteCoupon query.

Reply:
Resolved